### PR TITLE
🧹 Add tests that run the whole action

### PR DESCRIPTION
**Pull request by pje**
This adds some basic tests to confirm the behavior of running the whole action. I'm using Jest mocks for the Github API.

My Jest knowledge is pretty minimal. In particular the way I'm mocking the GitHub API feels janky. Feedback appreciated!

### CODE REVIEW

File: __tests__/main.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+const removeLabelMock = jest.spyOn(gh.issues, "removeLabel");
+const reposMock = jest.spyOn(gh.repos, "getContents");
+const paginateMock = jest.spyOn(gh, "paginate");
+const getPullMock = jest.spyOn(gh.pulls, "get");
+
+const yamlFixtures = {
+  "only_pdfs.yml": fs.readFileSync("__tests__/fixtures/only_pdfs.yml"),
+};
+
+function usingLabelerConfigYaml(fixtureName: keyof typeof yamlFixtures): void {
```
Comment (thboop) [inline __tests__/main.test.ts:21]: nit: for convention I would move these helper functions to the bottom of the file

File: __tests__/main.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    data: { content: yamlFixtures[fixtureName], encoding: "utf8" },
+  });
+}
+
+function mockGitHubResponseChangedFiles(...files: string[]): void {
+  const returnValue = files.map((f) => ({ filename: f }));
+  paginateMock.mockReturnValue(<any>returnValue);
+}
+
+afterAll(() => jest.restoreAllMocks());
```
Comment (thboop) [inline __tests__/main.test.ts:32]: Nit: consider setting the jest config value: https://jestjs.io/docs/configuration#restoremocks-boolean

Like we do for `clearMocks` https://github.com/actions/labeler/blob/main/jest.config.js#L2

**Review by thboop — APPROVED** [review]
LGTM

File: __tests__/main.test.ts (hunk 1 above)
Comment (pje) [reply __tests__/main.test.ts:21]: Wasy enough! Will do. 👍

**Comment by pje** [comment]
Thanks for all your thoughtful comments, @thboop!

**Pull request merged**